### DISCUSSION TITLE
Idea: no access to claims belonging to banned players

### DISCUSSION THREAD

**Discussion by bigpresh** [GriefPrevention/GriefPrevention#138] [Ideas]
Dropping this here as an idea I've not thought about in-depth yet - I'm not certain that it's (a) a good idea, (b) feasible without adding undue latency, but:

It would be nice if claims of banned players were rendered inaccessible to other players who previously had trust on them.  Use case: player A gets banned for x-raying, but player B has trust on their claim - player B can still come and retrieve the ill-gotten gains.  You could remove their trust, but if player A had a lot of claims, it could be an arse to go through.

(and, yes, player A could have given player B all the stuff beforehand, or stored it in player B's claim instead, but it would stop the most common case we see.)

I don't know if a lookup to find out if the claim owner is banned would be too slow to consider when checking whether to allow access to a claim or not.

**Comment by zedwick** [comment]
As some player legitimately share bases, and even claim their friend's builds with their own excess claim blocks, this could cause more problems than it solves.

However, a command to lock the specified player's claims could be particularly useful. You could use it with another plugin to automatically lock the players claims in the case of a ban for xray, or manually if you want to lock them temporarily whilst investigating. Having the command lock all current claims, but allow future claims to be accessible, would be better as this provides further scope for alternate gamemodes like hardcore where instead of old builds being deleted inside existing claims on death, they are instead locked to preserve their accomplishments in previous lives.

**Comment by bigpresh** [comment]
Hmm, yes, a command to lock/unlock all of a given player's claims could be a sensible way to go, and may be useful in other cases, too.

I agree that some players legitimately share bases, but IMO, if the base you're sharing is owned by a player banned for x-raying, it's not unreasonable to lose access to it - but a command to lock claims would allow it to be customised to the situation, rather than always denying access to claims of banned players, adding flexibility.
